cubeb-backend/src/log.rs

```diff
@@ -23,7 +23,8 @@ macro_rules! cubeb_log_internal {
     };
     (__INTERNAL__ $msg: expr) => {
         if let Some(log_callback) = $crate::ffi::g_cubeb_log_callback {
-            let cstr = ::std::ffi::CString::new(format!("{}:{}: {}\n", file!(), line!(), $msg)).unwrap();
+            let filename = std::path::Path::new(file!()).file_name().unwrap().to_str().unwrap();
+            let cstr = ::std::ffi::CString::new(format!("{}:{}: {}\n", filename, line!(), $msg)).unwrap();
```

Comment:
Is this also used for the `cubeb_logv` variant? `format!` allocates.

Reply:
I got severely nerd snipped by this it appears, here are my findings: https://github.com/padenot/perf-log/.

Reply:
https://github.com/padenot/perf-log/commit/a1797184e78412b4378f73970b518303b594361e gives 12% improvement but isn't required.